### ISSUE TITLE
Warn about redundant brackets in class constraints

### ISSUE THREAD

**Issue by asr** [ndmitchell/hlint#55]
In the following class definition

``` haskell
class (Eq a) => Foo a where
  foo :: a -> a
```

the  current development version  of `HLint` didn't warn about the redundant brackets in the constraint `(Eq a)`.

**Comment by ndmitchell** [comment]
Happy for people to contribute off-by-default hints, but if they aren't being actively worked on, I'm not planning to leave a ticket for them. If someone does want to actively work on it raise a ticket.
